tools/azure-sdk-tools/ci_tools/dependency_analysis.py

```diff
@@ -58,25 +58,36 @@ def record_dep(dependencies: Dict[str, Dict[str, Any]], req_name: str, spec: str
 def get_lib_deps(base_dir: str) -> Tuple[Dict[str, Dict[str, Any]], Dict[str, Dict[str, Any]]]:
     packages = {}
     dependencies = {}
-    for lib_dir in discover_targeted_packages("azure*", base_dir):
-        setup_path = os.path.join(lib_dir, "setup.py")
-        try:
-            parsed = ParsedSetup.from_path(setup_path)
-            lib_name, version, requires = parsed.name, parsed.version, parsed.requires
 
-            packages[lib_name] = {"version": version, "source": lib_dir, "deps": []}
+    def parse_setup(setup_path: str) -> None:
+        """Attempts to parse either a setup.py or pyproject.toml file to extract package information."""
+        parsed = ParsedSetup.from_path(setup_path)
+        lib_name, version, requires = parsed.name, parsed.version, parsed.requires
+
+        packages[lib_name] = {"version": version, "source": lib_dir, "deps": []}
 
-            for req in requires:
-                req_obj = parse_require(req)
-                req_name = req_obj.name
-                spec = req_obj.specifier if len(req_obj.specifier) else None
-                if spec is None:
-                    spec = ""
+        for req in requires:
+            req_obj = parse_require(req)
+            req_name = req_obj.name
+            spec = req_obj.specifier if len(req_obj.specifier) else None
+            if spec is None:
+                spec = ""
 
-                packages[lib_name]["deps"].append({"name": req_name, "version": str(spec)})
-                record_dep(dependencies, req_name, str(spec), lib_name)
+            packages[lib_name]["deps"].append({"name": req_name, "version": str(spec)})
+            record_dep(dependencies, req_name, str(spec), lib_name)
+
+    for lib_dir in discover_targeted_packages("azure*", base_dir):
+        setup_path = os.path.join(lib_dir, "setup.py")
```

Comment:
I recommend updating this to simply pass the lib_dir into the `ParsedSetup.from_path`. [This code](https://github.com/Azure/azure-sdk-for-python/blob/83ebd8bfeb0712bfa208a7476587c671d79f6951/tools/azure-sdk-tools/ci_tools/parsing/parse_functions.py#L600) specifically is why.

Reply:
Great call-out! Updated.